javalib/src/main/scala/java/util/Formatter.scala

```diff
@@ -383,8 +383,8 @@ final class Formatter private (private[this] var dest: Appendable,
             padAndSendToDest(RootLocaleInfo, flags, width, prefix,
                 java.lang.Long.toOctalString(arg))
           case arg: BigInteger =>
-            padAndSendToDest(RootLocaleInfo, flags, width, prefix,
-              applyNumberUpperCase(flags, arg.toString(8)))
+            formatNumericString(RootLocaleInfo, flags, width,
```

Comment:
If we're attempting to use `formatNumericString` for octal and hexadecimal, we must add a test in `FormatterTestEx` in the method that tests devanagari digits. I am pretty sure things are going to go wrong there, because the localization algorithm will be applied, but it should not.

If `formatNumericString` doesn't do the right thing out of the box, then we shouldn't use it. The whole point of that method is to apply the localization algorithm. We may need something else, closer to the bare `padAndSendToDest`, but which takes care of the `-` prefix.